pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1075,3 +1101,96 @@ func (c *K8sOrchestrator) GetNodesForVolumes(ctx context.Context, volumeIDs []st
 	}
 	return volumeIDToNodeNames
 }
+
+// initNodeIDToNameMap performs all the operations required to initialize
+// the node ID to  name map. It also watches for node add, update & delete
+// operations, and updates the map accordingly.
+func initNodeIDToNameMap(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+
+	log.Debugf("Initializing node ID to node name map")
+	k8sOrchestratorInstance.nodeIDToNameMap = &nodeIDToNameMap{
+		RWMutex: &sync.RWMutex{},
+		items:   make(map[string]string),
+	}
+
+	// Set up kubernetes resource listener to listen events on Node
+	k8sOrchestratorInstance.informerManager.AddNodeListener(
+		func(obj interface{}) { // Add.
+			nodeAdd(obj)
+		},
+		nil,
+		func(obj interface{}) { // Delete.
+			nodeRemove(obj)
+		})
+}
+
+// nodeAdd adds an entry into nodeIDToNameMap. The node MoID is retrieved from the
+// node annotation vmware-system-esxi-node-moid
+func nodeAdd(obj interface{}) {
+	log := logger.GetLogger(context.Background())
+	node, ok := obj.(*v1.Node)
+	if node == nil || !ok {
+		log.Warnf("nodeAdd: unrecognized object %+v", obj)
+		return
+	}
+
+	log.Debugf("nodeAdd: node=%v", node)
+	nodeMoID, ok := node.ObjectMeta.Annotations[common.HostMoidAnnotationKey]
+	if !ok {
+		log.Debugf("nodeAdd: %v annotation not found on the node %v", common.HostMoidAnnotationKey, node.Name)
+		return
+	}
+	k8sOrchestratorInstance.nodeIDToNameMap.add(nodeMoID, node.Name)
+}
+
+// nodeRemove removes an entry from nodeIDToNameMap. The node MoID is retrieved from the
+// node annotation vmware-system-esxi-node-moid
+func nodeRemove(obj interface{}) {
+	log := logger.GetLogger(context.Background())
+	node, ok := obj.(*v1.Node)
+	if node == nil || !ok {
+		log.Warnf("nodeRemove: unrecognized object %+v", obj)
+		return
+	}
+
+	log.Debugf("nodeRemove: node=%v", node)
+	nodeMoID, ok := node.ObjectMeta.Annotations[common.HostMoidAnnotationKey]
+	if !ok {
+		log.Debugf("nodeRemove: %v annotation not found on the node %v", common.HostMoidAnnotationKey, node.Name)
```

Comment:
Can you use format specifiers in a consistent way. Sometimes we are using %v & other places I observed %s

Reply:
I have updated strings to use %s